coremltools/models/utils.py

```diff
@@ -1035,25 +1062,31 @@ def compile_model(model: _Union['_ct.models.MLModel', str, _Model_pb2.Model]) ->
     except:
         raise Exception("Unable to compile any Core ML models.")
 
-    # Check parameter
-    if not isinstance(model, (str, _Model_pb2.Model, _ct.models.MLModel)):
-        raise Exception("Compiling a Core ML models is only support on macOS 10.13 or higher.")
+    # Check model parameter
+    if isinstance(model, str):
+        raise Exception("To get a compiled model from a saved MLModel, first load the model, "
```

Comment:
nit: Try to use more specific exception to avoid the general Exception as suggested by PEP8. I would recommend `ValueError` for invalid parameters in util functions.